root/programs/C#/Frameworks/Infrastructure/Public/Util/CodeSigningX509.cs

```diff
@@ -0,0 +1,198 @@
+
+        /// <summary>X.509証明書の秘密鍵</summary>
+        public string X509PrivateKey
+        {
+            get
+            {
+                return "- hidden -";
+            }
+        }
+
+        /// <summary>X.509証明書の公開鍵</summary>
+        public string X509PublicKey
+        {
+            get
+            {
+                return CustomEncode.ToBase64String(this._X509Certificate.GetPublicKey());
+            }
+        }
+
+        /// <summary>
+        /// Constructor
+        /// X.509証明書(*.pfx, *.cer)からキーを設定する。
+        /// *.cer証明書の場合は、証明書チェーンが繋がっている必要がある。
+        /// 自己証明書の場合「信頼されたルート証明機関」にInstallするなどする。
+        /// </summary>
+        /// <param name="certificateFilePath">X.509証明書(*.pfx, *.cer)へのパス</param>
+        /// <param name="password">パスワード</param>
+        /// <param name="hashAlgorithmName">
+        /// RSAの場合、以下からHashAlgorithm名を指定する。
+        /// MD5, SHA1, SHA256, SHA384, SHA512
+        /// </param>
+        public CodeSigningX509(string certificateFilePath, string password, string hashAlgorithmName)
+        {
+            this._X509Certificate = new X509Certificate2(
+                certificateFilePath, password, X509KeyStorageFlags.Exportable);
+
+            this._hashAlgorithmName = hashAlgorithmName;
+        }
+
+        #endregion
+
+        #region 署名(X509Certificate)
+
+        /// <summary>Sign</summary>
+        /// <param name="data">data</param>
+        /// <returns>署名</returns>
+        public override byte[] Sign(byte[] data)
+        {
+            // アルゴリズム（RSA or DSA）は、ココで決まるもよう。
+            AsymmetricAlgorithm aa = this._X509Certificate.PrivateKey;
+
+            // 署名
+            byte[] signedByte = null;
+
+            if (aa is RSACryptoServiceProvider)
+            {
+                // RSACryptoServiceProvider : ExportParametersして生成し直している。
+                RSAParameters rsaPara = ((RSACryptoServiceProvider)(aa)).ExportParameters(true);
+                RSACryptoServiceProvider rsa = new RSACryptoServiceProvider(this._X509Certificate.PrivateKey.KeySize);
+                rsa.ImportParameters(rsaPara);
+                signedByte = rsa.SignData(data, this._hashAlgorithmName);
+            }
+            else
+            {
+                // DSACryptoServiceProvider : ExportParametersして生成し直している。
+                DSAParameters dsaPara = ((DSACryptoServiceProvider)(aa)).ExportParameters(true);
+                DSACryptoServiceProvider dsa = new DSACryptoServiceProvider(this._X509Certificate.PrivateKey.KeySize);
+                dsa.ImportParameters(dsaPara);
+                signedByte = dsa.SignData(data); // SHA-1 固定のため？
+            }
+
+            return signedByte;
+        }
+
+        /// <summary>Verify</summary>
+        /// <param name="data">data</param>
+        /// <param name="sign">署名</param>
+        /// <returns>検証結果</returns>
+        public override bool Verify(byte[] data, byte[] sign)
+        {
+            AsymmetricAlgorithm aa = null;
+
+            // 結果フラグ
```

Comment:
@daisukenishino 

Please show the meaning of this flag.
Does *true* mean *success*?
Does *false* mean *failure*?